### ISSUE TITLE
Bug: impl block with generics

### ISSUE THREAD

**Issue by Banyc** [elastio/bon#16]
```rust
pub struct G<T> {
    buf: Vec<T>,
}
#[bon::bon]
impl<T> G<T> {
    #[builder]
    pub fn new() -> Self {
        Self { buf: vec![] }
    }
}
```

Error msg:

```text
error[E0392]: type parameter `T` is never used
 --> src/lib.rs:5:6
  |
5 | impl<T> G<T> {
  |      ^
  |      |
  |      unused type parameter
  |      `T` is named here, but is likely unused in the containing type
  |
  = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`
  = help: if you intended `T` to be a const parameter, use `const T: /* Type */` instead

error[E0392]: type parameter `T` is never used
 --> src/lib.rs:5:6
  |
5 | impl<T> G<T> {
  |      ^ unused type parameter
  |
  = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`
  = help: if you intended `T` to be a const parameter, use `const T: /* Type */` instead

error[E0282]: type annotations needed
 --> src/lib.rs:4:1
  |
4 | #[bon::bon]
  | ^^^^^^^^^^^ cannot infer type
  |
  = note: this error originates in the attribute macro `bon::bon` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0282, E0392.
For more information about an error, try `rustc --explain E0282`.
```

**Comment by Banyc** [comment]
Solution is probably to add the generics `T` to the phantom; illustrated by the selected part:

<img width="709" alt="image" src="https://github.com/user-attachments/assets/28267c2f-3be4-4d0b-bc3c-d7a6bba50f12">

**Comment by Veetaha** [comment]
Thank you for the bug report! The fix was implemented in https://github.com/elastio/bon/pull/17. A new patch version of the crate `1.0.3` with the fix was released.

Feel free to reopen the issue if the problem persists

**Comment by Veetaha** [comment]
Btw, @Banyc, when looking at your screenshot I found a bug that the method `new()` isn't hidden by default. I'm not sure if this bug affects your code. It doesn't if you don't call the `new()` method directly.

The method `new()` was supposed to be renamed to `__orig_new()` and `#[doc(hidden)]` should've been added to it. If you'd like to have both `new()` and `builder()` methods available, then you should explicitly state that with [`#[builder(expose_positional_fn)]`](https://elastio.github.io/bon/docs/reference/builder#expose-positional-fn).

I prepared a fix for this other bug in #19 that I'd like to release as a new  1.0.4 patch.
